pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -628,13 +628,16 @@ private File downloadAndStreamUntarWithRateLimit(String segmentName, SegmentZKMe
         maxStreamRateInByte);
     String uri = zkMetadata.getDownloadUrl();
     try {
-      File ret = SegmentFetcherFactory.fetchAndStreamUntarToLocal(uri, tempRootDir, maxStreamRateInByte);
-      LOGGER.info("Download and untarred segment: {} for table: {} from: {}", segmentName, _tableNameWithType, uri);
-      return ret;
-    } catch (AttemptsExceededException e) {
-      LOGGER.error("Attempts exceeded when stream download-untarring segment: {} for table: {} from: {} to: {}",
-          segmentName, _tableNameWithType, uri, tempRootDir);
-      _serverMetrics.addMeteredTableValue(_tableNameWithType, ServerMeter.SEGMENT_STREAMED_DOWNLOAD_UNTAR_FAILURES, 1L);
+        File ret = SegmentFetcherFactory.fetchAndStreamUntarToLocal(uri, tempRootDir, maxStreamRateInByte);
+        LOGGER.info("Downloaded and untarred segment: {} for table: {} from: {}", segmentName, _tableNameWithType, uri);
+        return ret;
+    } catch (Exception e) {
```

Comment:
Hey @jugomezv, I might miss something here, but if our purpose here is to add to a metric every time a download-untar fails, then adding to the counter here in this catch block will not capture all the failures? If we want to have best observability and capture all the failures that happens even in the attempts, should we do that in https://github.com/apache/pinot/blob/2050376adcb67041972d9707b338373a5e6bc468/pinot-common/src/main/java/org/apache/pinot/common/utils/fetcher/HttpSegmentFetcher.java#L109

Reply:
Yes I think you are correct, this will likely catch only permanent errors likely missed all the others we handle specifically there, let me move

Reply:
Discussed further and we decided to let BaseRetryPolicy.attempt() return the number of failure attempts.